.github/COMMIT_TEMPLATE

```diff
@@ -0,0 +1,44 @@
+# Short (50 chars or less) summary of change
+# "If this commit is applied then this commit will XYZ"
+# where *XYZ* is the 50 chars summary
+
+
+# Detailed summary goes below in the form of an
+# opinionated Q&A
+# bullet points preferred over paragraphs
+
+1. Why is this change necessary ?
+# Eg.
+# - improves X by implementing feature A in B
+# - reduces latency between P & Q
+# <should have enough detail>
+
+2. How does this change address the issue ?
+# Eg.
+# - mitigates case Y in Z machines.
+# - avoids situation X by a code fix
+
+
+3. How to verify this change ?
+# Eg.
+# - command Z does not give error and exits
```

Comment:
This should be as detailed as possible. The reviewer of a commit should be able to setup, install, update & run the commands to check if the commit actually works. The reviewer may not necessarily do it but this is essential.

There will be e2e test cases which will be dependent on these steps.

Reply:
I've added `<should be most detailed>`

Reply:
- a sample one which you can work out better using your imaginations

```
- setup XYZ 2.0.0
- Run command abc
- Run command def
- Verify the output as 123
```